action.yml

```diff
@@ -18,10 +22,26 @@ runs:
     - uses: actions/setup-go@v3
       with:
         go-version: ${{ inputs.go_version }}
+
+    # if GITHUB_TOKEN env var is set, use it
+    - id: choose_token
+      run: |
+        if [ -n "$GITHUB_TOKEN" ]; then
+          token=$GITHUB_TOKEN
+        else
+          token=$REPO_TOKEN
+        fi
+        echo "TOKEN=$token" >> "$GITHUB_OUTPUT"
+      env:
+        REPO_TOKEN: ${{ github.token }}
+      shell: bash
+
```

Comment:
**Would it be cleaner to have an input for an explicit token, falling back onto the automatic token if one isn't provided?**

This isn't too much of a departure of what is going on here except that it uses an input rather than an environment variable.  An explicit input parameter would be a stronger and more readily seen contract than an environment variable.

Reply:
Yes, absolutely. I was actually thinking the same as I was writing it. I'll change it to an input parameter!